compat/statvfs.cc

```diff
@@ -21,7 +21,7 @@
 int
 xstatvfs(const char *path, struct statvfs *sfs)
 {
-#if !HAVE_STATFS && _SQUID_WINDOWS_
+#if !HAVE_STATFS && _SQUID_MINGW_
```

Comment:
This "breaks" VisualStudio builds. But that environment has not built since Squid-2.7, with no support in sight it is effectively dead. So, I am switching these code blocks that are useful to MinGW.  When the MinGW port is working we can decide whether to drop the remainder VisualStudio-only code or not.

Reply:
In relation to PR #1085, we should copy the fixes from that PR code to this new `os/mingw.h` (when proven needed) instead of from the broken master branch `os/mswindows.h` file.

Reply:
> Why not remove all VisualStudio-only code as a starting point?

Several reasons:
 * I do not want to erase a lot of code only to re-add it.
 * Someone on the review team has a history of you rejecting such follow PRs on grounds of the re-addition being "new code". I do not want to spend years arguing over code style/logic/documentation which is currently status-quo just to get Squid to build again using the code it already has today.

Reply:
Reverted the removal in [ba0e1ea](https://github.com/squid-cache/squid/pull/1341/commits/ba0e1ea54eb845c4fb7326f3658b316fa3a94dae)